### DISCUSSION TITLE
Does commitizen use any python-based cli framework?

### DISCUSSION THREAD

**Discussion by simkimsia** [commitizen-tools/commitizen#860] [Q&A] [answered]
As a user, I enjoy using commitizen even though it's been merely a week since I found it.

I was wondering if the team built it from some existing python-based cli framework.

Here's a list of well-known python-based cli frameworks. https://github.com/shadawck/awesome-cli-frameworks#python. I wonder if it uses any or some of the stated frameworks.

**Comment by woile** [comment, marked as answer]
We use argparse directly through [decli](https://github.com/woile/decli).

Internally it uses [questionary](https://questionary.readthedocs.io/en/stable/) for the nice looking prompts
